server/src/test/java/org/elasticsearch/env/NodeEnvironmentTests.java

```diff
@@ -627,10 +622,13 @@ public void testIndexCompatibilityChecks() throws IOException {
                 () -> checkForIndexCompatibility(logger, env.dataPaths())
             );
 
+            // TODO[wrb] replace range with "pre-7.x" or something
```

Comment:
If the release version string is incorrect, we should fix it in the `toReleaseVersion` method

Reply:
Once again, I think it's the range that's a little confusing here. For IDs that are before the earliest known version, the upper bound seems to come back as 7.0.0, which leads to ranges like `5.12.2-7.0.0`. Just returning `5.12.2` seems wrong, because we might confidently tell people to downgrade to a nonexistent version... for this PR I think I'll just assume that what comes back from `toReleaseVersion` is what we want to show, and we can refine that output elsewhere.